fredericoo/saltplay
# fix: migration on preview and prod environments

# Overview

Fixing a long running problem with migrations needing manual copy-pasting of database connection strings, and also automating staging db pushes.

## What we have done

- Defaulted some env variables at `.env.development` to reduce friction when running the app on a new machine
- Created github action that automatically migrates the preview database to match the latest version of the `prisma/migrations` folder
- Set default formatter for `prisma` files (so the schema looks pretty without us needing to tab tab tab)
- Reformatted (automatically) our schema file with standardised indentation etc
- Split `db:migrate` into local, prev and prod environments
- On dev environments that have `NEXT_PUBLIC_ENABLE_DEV_LOGIN` as true, if no users are found, now a user called `admin` is created to make it possible to access the dashboard.

## How to test

It's a bit difficult to test, but here we go:

- Modify the schema by changing the `schema.prisma` file. Create a new field, to be on the safer side.
- Run the migration command locally `yarn db:migrate:local` to create a migration
- Commit this migration
- Under Checks on this PR (at the bottom of this page), you should see that it did *not* skip the Install Packages and Run Migration steps.

## Checklist:

- I have created automated tests to replicate my manual testing automatically
